ui/button.mod/button.js

```diff
@@ -1,485 +1,470 @@
+     * @param {MutableEvent} mutableEvent - The event object
+     * @protected
+     * @fires action
+     */
+    handleKeyPress(mutableEvent) {
+        const { identifier } = mutableEvent;
+
+        if (identifier === "space" || identifier === "enter") {
+            this.active = false;
+            this.dispatchActionEvent();
         }
-    },
-
-    handleLongPress: {
-        value: function(event) {
-            if (!this._promise){
-                // When we fire the "hold" event we don't want to fire the
-                // "action" event as well.
-                this._pressComposer.cancelPress();
-                this._removeEventListeners();
-
-                var longActionEvent = document.createEvent("CustomEvent");
-                longActionEvent.initCustomEvent("longAction", true, true, null);
-                this.dispatchEvent(longActionEvent);
-            }
+    }
+
+    /**
+     * Called when the user starts interacting with the component.
+     * @protected
+     * @param {MutableEvent} mutableEvent - The event object
+     */
+    handlePressStart(mutableEvent) {
+        if (!this._promise) {
+            this.active = true;
+            this._addEventListeners();
         }
-    },
+    }
 
     /**
-    Called when all interaction is over.
-    @private
-    */
-    handlePressCancel: {
-        value: function(event) {
+     * Called when the user has interacted with the button.
+     * @protected
+     * @param {MutableEvent} mutableEvent - The event object
+     * @fires action
+     */
+    handlePress(mutableEvent) {
+        if (!this._promise) {
             this.active = false;
+            this.dispatchActionEvent(event.details);
             this._removeEventListeners();
         }
-    },
+    }
 
     /**
-    If this is an input element then the label is handled differently.
-    @private
-    */
-    _isInputElement: {
-        value: undefined,
-        enumerable: false
-    },
-    isInputElement: {
-        get: function() {
-            return this._isInputElement !== undefined ? this._isInputElement : (this._isInputElement = (this.element ? (this.element.tagName === "INPUT") : false));
-        },
-        enumerable: false
-    },
-
-    enterDocument: {
-        value: function (firstDraw) {
-            if (Control.prototype.enterDocument) {
-                Control.prototype.enterDocument.apply(this, arguments);
-            }
+     * Called when the user has interacted with the button for a long time.
+     * @protected
+     * @param {MutableEvent} mutableEvent - The event object
+     * @fires longAction
+     */
+    handleLongPress(mutableEvent) {
+        if (!this._promise) {
+            // When we fire the "hold" event we don't want to fire the
+            // "action" event as well.
+            this._pressComposer.cancelPress();
+            this._removeEventListeners();
 
-            if (firstDraw) {
```

Comment:
As discussed, restore this logic and remove the text component from the template.